geoportal/c2cgeoportal_geoportal/views/entry.py

```diff
@@ -197,10 +197,24 @@ def _get_metadatas(self, item, errors):
     def _wms_getcap(self, ogc_server=None):
         ogc_server = (ogc_server or self.default_ogc_server)
 
-        return self._wms_getcap_cached(
+        url, content, errors = self._wms_getcap_cached(
             ogc_server, self._get_capabilities_cache_role_key(ogc_server)
         )
 
+        if len(errors) != 0:
+            return None, errors
+
+        wms = None
+        try:
+            wms = WebMapService(None, xml=content)
+        except Exception as e:  # pragma: no cover
+            error = "WARNING! an error occurred while trying to read the mapfile and recover the themes." \
+                "\nURL: {}\n{}".format(url, content)
+            errors.add(error)
+            log.error(error)
+            log.exception(e)
```

Comment:
You should never do that. Just just add `exc_info=True` to the previous line.

Reply:
Thanks, fixed :-)